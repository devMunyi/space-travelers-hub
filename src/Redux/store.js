import { configureStore, applyMiddleware } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import rocketsReducers from './reducers/rockets';

export default configureStore(
  {
    reducer: {
      rockets: rocketsReducers,
    },
  },
  applyMiddleware(logger)
);
