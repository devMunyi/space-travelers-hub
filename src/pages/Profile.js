import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchRockets } from '../redux/actions/rockets';
import { fetchMissions } from '../redux/mission/missions';

const Profile = () => {
  // set dispatch
  const dispatch = useDispatch();

  // fetch rockets on page mount
  useEffect(() => {
    dispatch(fetchRockets());
  }, [dispatch]);

  // fetch missions on page mount
  useEffect(() => {
    dispatch(fetchMissions());
  }, [dispatch]);

  // retrieve rockets from store for rendering
  const rockets = useSelector((state) => state.rockets);

  // retrieve missions from store for rendering
  const missions = useSelector((state) => state.missions);

  return (
    <div className="container">
      <div className="row">
        <div className="col-md-6">
          <table className="table table-bordered">
            <thead>
              <tr>
                <th scope="col">My Missions</th>
              </tr>
            </thead>
            <tbody>
              {missions.map((mission) => (
                <tr key={mission.id}>
                  <td>{mission.mission_name}</td>
                </tr>
              ))}
              {/* <tr>
                <td>Mark</td>
              </tr>
              <tr>
                <td>Thornton</td>
              </tr>
              <tr>
                <td>the Bird</td>
              </tr> */}
            </tbody>
          </table>
        </div>
        <div className="col-md-6">
          <table className="table table-bordered">
            <thead>
              <tr>
                <th scope="col">My Rockets</th>
              </tr>
            </thead>
            <tbody>
              {rockets.map((rocket) => (
                <tr key={rocket.id}>
                  <td>{rocket.rocket_name}</td>
                </tr>
              ))}
            </tbody>
          </table>
        </div>
      </div>
    </div>
  );
};

export default Profile;
