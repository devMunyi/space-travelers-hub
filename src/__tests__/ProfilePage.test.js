import React from 'react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Profile from '../pages/Profile';
import store from '../redux/store';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Profile />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
