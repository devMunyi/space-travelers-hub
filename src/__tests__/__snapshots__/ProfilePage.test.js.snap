// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-6"
    >
      <table
        className="table table-bordered"
      >
        <thead>
          <tr>
            <th
              scope="col"
            >
              My Missions
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <div
      className="col-md-6"
    >
      <table
        className="table table-bordered"
      >
        <thead>
          <tr>
            <th
              scope="col"
            >
              My Rockets
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</div>
`;
